test/suite/test_compact10.py

```diff
@@ -0,0 +1,161 @@
+        cursor = self.session.open_cursor('backup:', None, None)
+        os.mkdir(todir)
+        while True:
+            ret = cursor.next()
+            if ret != 0:
+                break
+            bkup_file = cursor.get_key()
+            copy_file = os.path.join(fromdir, bkup_file)
+            shutil.copy(copy_file, todir)
+        assert ret == wiredtiger.WT_NOTFOUND
+        cursor.close()
+
+    def turn_on_bg_compact(self, config):
+        self.session.compact(None, config)
+        compact_running = self.get_bg_compaction_running()
+        while not compact_running:
+            time.sleep(1)
+            compact_running = self.get_bg_compaction_running()
+        self.assertEqual(compact_running, 1)
+
+    def test_compact10(self):
+
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        # Create and populate tables.
+        uris = []
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            uris.append(uri)
+            self.session.create(uri, self.create_params)
+            self.populate(uri, self.table_numkv, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete 50% of the file.
+        for uri in uris:
+            self.delete_range(uri, 50 * self.table_numkv // 100)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Reconfigure the connection with:
+        # - Aggressive compaction
+        reconf_config = self.conn_config + ',debug_mode=(background_compact)'
+        # - Stressing options to slow down compaction and checkpoints so compaction takes more time.
+        reconf_config += f',timing_stress_for_test=({self.timing_stress_cfg})'
+        # - Verbose for debugging purposes.
+        # reconf_config += ',verbose=(compact,compact_progress)'
+        self.conn.reconfigure(reconf_config)
+        
+        # Enable background compaction.
+        compact_config = f'background=true,free_space_target=1MB'
+        self.turn_on_bg_compact(compact_config)
+
+        # Wait for a random table to be compacted before crashing.
+        uri_idx = random.randint(0, self.n_tables - 1)
+        uri = uris[uri_idx]
+        # Given 50% of each collection has been deleted, compaction must have work to do.
+        while not self.get_pages_rewritten(uri):
+            time.sleep(0.5)
+
+        # Sleep some random time before proceeding.
+        time.sleep(random.uniform(0, 0.5))
+
+        # Take a full backup.
+        self.take_full_backup('.', self.backup_dir)
+        self.reopen_conn(self.backup_dir, self.conn_config)
+
+        # Verify each table from the backup.
+        for uri in uris:
+            self.verifyUntilSuccess(self.session, uri, "strict")
+
+        # Background compaction may be have been inspecting a table when disabled which is
```

Comment:
Is this during connection close?

Reply:
Yes, when we don't turn off compaction explicitly, the framework will close the connection and the background compaction may say it was in the middle of something.

Reply:
Okay got it, thanks